torchvision/prototype/transforms/_misc.py

```diff
@@ -225,28 +226,99 @@ def _transform(
-            bounding_box.as_subclass(torch.Tensor),
-            old_format=bounding_box.format,
+        self.labels = labels
+        if labels == "default":
+            self._get_labels = self._find_label_default_heuristic
+        elif callable(self.labels):
+            self._get_labels = labels
+        elif isinstance(self.labels, str):
+            self._get_labels = lambda inputs: inputs[labels]
+        else:
+            raise ValueError(
+                "labels parameter should either be a str, callable, or 'default'. "
+                f"Got {labels} of type {type(labels)}."
+            )
+
+    def _find_label_default_heuristic(self, inputs):
+        # Tries to find a "label" key, otherwise tries for the first key that contains "label" - case insensitive
+        # Returns None if nothing is found
+        labels = None
+        candidate_key = None
+        with suppress(StopIteration):
+            candidate_key = next(key for key in inputs.keys() if key.lower() == "labels")
+        if candidate_key is None:
+            with suppress(StopIteration):
+                candidate_key = next(key for key in inputs.keys() if "label" in key.lower())
+        labels = inputs.get(candidate_key)
+
+        if labels is None:
+            raise ValueError(
+                "Could not infer where the labels are in the sample. Try passing a callable as the label parameter?"
+            )
+        return labels
+
+    def forward(self, *inputs: Any) -> Any:
+        inputs = inputs if len(inputs) > 1 else inputs[0]
+
+        if isinstance(self.labels, str) and not isinstance(inputs, dict):
+            raise ValueError(
+                f"If labels is a str or 'default' (got {self.labels}), then the input to forward() must be a dict. "
+                f"Got {type(inputs)} instead"
+            )
+        labels = self._get_labels(inputs)
+        if not isinstance(labels, torch.Tensor):
+            raise ValueError(f"The labels in the input to forward() must be a tensor, got {type(labels)} instead.")
+
+        flat_inputs, spec = tree_flatten(inputs)
+        # TODO: this enforces one single BoundingBox entry.
+        # Assuming this transform needs to be called at the end of *any* pipeline that has bboxes...
+        # should we just enforce it for all transforms?? What are the benefits of *not* enforcing this?
+        boxes = query_bounding_box(flat_inputs)
+
+        if boxes.shape[-2] != labels.shape[0]:
+            raise ValueError(
+                f"Number of boxes ({boxes.shape[-2]}) and number of labels ({labels.shape[0]}) do not match."
+            )
+
+        boxes = F.convert_format_bounding_box(
+            boxes,
             new_format=datapoints.BoundingBoxFormat.XYXY,
         )
-        valid_indices = remove_small_boxes(bounding_box, min_size=self.min_size)
-
-        return dict(valid_indices=valid_indices)
+        ws, hs = boxes[:, 2] - boxes[:, 0], boxes[:, 3] - boxes[:, 1]
```

Comment:
Thinking about this more I don't think it is possible to support multiple batches. Say we remove 3 boxes in the first batch entry and 2 boxes in the second batch entry: we end up with a batch with entries of different size - that's not supported.

I'll still add support for batch with 1 element, just in case (and error otherwise)